core/src/protocols_handler/mod.rs

```diff
@@ -134,8 +134,25 @@ pub trait ProtocolsHandler {
     /// therefore no more inbound substreams will be produced.
     fn inject_inbound_closed(&mut self);
 
+    /// Returns whether the connection should be kept alive.
+    ///
+    /// If returns `false`, that indicates that this connection is not important and the user may
+    /// invoke `shutdown()` if they think that they will no longer need the connection in the
+    /// future.
+    ///
+    /// On the other hand, returning `true` is only an indication and doesn't mean that the user
+    /// will not call `shutdown()`.
+    ///
+    /// When multiple `ProtocolsHandler` are combined together, they should use *AND* to merge the
```

Comment:
I am confused. The PR description states:

> A connection will always stay alive any of the handlers returns `true` for `connection_keep_alive()`.

So why`AND` here and [here](https://github.com/libp2p/rust-libp2p/pull/816/files#diff-4685ea3ff5643713fad0b2edf588e2b2R160)?

Reply:
Stupidity.